test/flycheck-test.el

```diff
@@ -3881,6 +3881,33 @@ Why not:
     (flycheck-ert-should-syntax-check
      "language/rust/lib-main/src/main.rs" 'rust-mode)))
 
+(flycheck-ert-def-checker-test rust-cargo rust notes
+  ;; We only get notes on the first build. Ensure we start from a
+  ;; clean directory each time.
+  (call-process-shell-command
```

Comment:
Haha, I concede that `shell-command` is less scary and a bit more readable.  I had the same reservations as you did at first.  Yet, `call-process` is the right  choice here.  After a while you don't see the `nil` (just like you don't see the parens!).